src/core/utils/channel_monitor.cpp

```diff
@@ -122,8 +122,14 @@ void ChannelMonitor::HandleTimer(Timer &aTimer)
 
 void ChannelMonitor::HandleTimer(void)
 {
-    GetInstance().Get<Mac::Mac>().EnergyScan(mScanChannelMasks[mChannelMaskIndex], 0,
-                                             &ChannelMonitor::HandleEnergyScanResult, this);
+    Instance &instance = GetInstance();
+
+    if (instance.Get<Mle::MleRouter>().GetRole() != OT_DEVICE_ROLE_DISABLED)
+    {
+        instance.Get<Mac::Mac>().EnergyScan(mScanChannelMasks[mChannelMaskIndex], 0,
+                                            &ChannelMonitor::HandleEnergyScanResult, this);
+    }
+
     RestartTimer();
```

Comment:
I think this timer restart may still pose a problem.  During Lurk it will cause the NCP to wake up thereby consuming more power.